adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2ADAM.scala

```diff
@@ -60,8 +56,8 @@ class Vcf2ADAM(val args: Vcf2ADAMArgs) extends BDGSparkCommand[Vcf2ADAMArgs] wit
   def run(sc: SparkContext) {
 
     val variantContextRdd = sc.loadVcf(args.vcfPath)
-    var variantContextsToSave = if (args.coalesce > 0) {
-      if (args.coalesce > variantContextRdd.rdd.partitions.size || args.forceShuffle) {
+    val variantContextsToSave = if (args.coalesce > 0) {
+      if (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle) {
```

Comment:
We can cut this if then else out:
`variantContextRdd.transform(_.coalesce(args.coalesce, shuffle = (args.coalesce > variantContextRdd.rdd.partitions.length || args.forceShuffle)))`